src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,63 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+
+    private int supply;
+    private int buy;
+    private boolean isParsed = false;
+
+    private int getResult() {
+        return supply - buy;
+    }
+
     public void getStatistic(String fromFileName, String toFileName) {
+        if (!isParsed) {
+            parse(fromFileName);
+            isParsed = true;
+        }
+        makeReport(toFileName);
+    }
 
+    public void parse(String fileName) {
```

Comment:
this method is named `parse`, although you read file contents here and then calculate a report

therefore method name lies about method behavior

I would recommend splitting this method into two parts - `readFile` and `calculateReport`